lib/shared/sdk/usbboot/index.js

```diff
@@ -158,230 +133,284 @@ const usbIdToString = (id) => {
+ * @returns {Boolean} whether the device is usbboot-capable
  *
  * @example
- * const usb = require('usb')
- * const device = usb.findByIds(0x0a5c, 0x2763)
- * const bootcode = fs.readFileSync('./bootcode.bin')
- *
- * writeBootCode(device, device.interfaces(0).endpoint(1), bootcode).then(() => {
- *   console.log('Done!')
- * })
+ * if (isUsbBootCapableUSBDevice({ ... })) {
+ *   console.log('We can use usbboot on this device')
+ * }
  */
-const writeBootCode = (device, endpoint, bootCodeBuffer) => {
-  debug('Writing bootcode')
-  debug(`Bootcode buffer length: ${bootCodeBuffer.length}`)
-  const bootMessageBuffer = protocol.createBootMessageBuffer(bootCodeBuffer.length)
-
-  debug('Writing boot message buffer to out endpoint')
-  return protocol.write(device, endpoint, bootMessageBuffer).then(() => {
-    debug('Writing boot code buffer to out endpoint')
-    return protocol.write(device, endpoint, bootCodeBuffer)
-  }).then(() => {
-    debug('Reading return code from device')
-    return protocol.read(device, protocol.RETURN_CODE_LENGTH)
-  }).then((data) => {
-    const returnCode = data.readInt32LE()
-    debug(`Received return code: ${returnCode}`)
-
-    if (returnCode !== protocol.RETURN_CODE_SUCCESS) {
-      throw new Error(`Couldn't write the bootcode, got return code ${returnCode} from device`)
-    }
+const isUsbBootCapableUSBDevice = (device) => {
+  return _.some(USBBOOT_CAPABLE_USB_DEVICES, {
+    vendorID: device.deviceDescriptor.idVendor,
+    productID: device.deviceDescriptor.idProduct
   })
 }
 
 /**
- * @summary Mount a USB device as a block device (usbboot second stage)
- * @function
- * @private
- *
- * @description
- * The possible files you can pass here are:
- *
- * - autoboot.txt
- * - config.txt
- * - recovery.elf
- * - start.elf
- * - fixup.dat
- *
- * @param {Object} device - node-usb device
- * @param {Object} endpoint - node-usb endpoint
- * @param {Object} files - a set of file buffers
- * @returns {Promise}
- *
- * @example
- * const usb = require('usb')
- * const device = usb.findByIds(0x0a5c, 0x2763)
- *
- * startFileServer(device, device.interfaces(0).endpoint(1), {
- *   'start.elf': fs.readFileSync('./start.elf')
- * }).then(() => {
- *   console.log('Done!')
- * })
+ * @summary StandardAdapter
+ * @class
  */
-const startFileServer = (device, endpoint, files) => {
-  debug('Listening for file messages')
-  return protocol
-    .read(device, protocol.FILE_MESSAGE_SIZE)
-    .then(protocol.parseFileMessageBuffer)
-
-    // We get these error messages when reading a command
-    // from the device when the communication has ended
-    .catch({
-      message: 'LIBUSB_TRANSFER_STALL'
-    }, {
-      message: 'LIBUSB_TRANSFER_ERROR'
-    }, (error) => {
-      debug(`Got ${error.message} when reading a command, assuming everything is done`)
-      return {
-        command: protocol.FILE_MESSAGE_COMMANDS.DONE
+class USBBootAdapter extends EventEmitter {
+  /**
+   * @summary StandardAdapter constructor
```

Comment:
Typo here? Should it be `USBBootAdapter constructor` ?

Reply:
Whoops. Fixed!